woocommerce/pinterest-for-woocommerce
# Track Save Button Click

### Changes proposed in this Pull Request:

Closes part of #232 

save_changes_button_click | Triggered when save changes button is clicked https://d.pr/i/4CiZgY | context: e.g. pinterest_settings or pinterest_connectionindication of what is being saved/changed 
-- | -- | -- 


### Screenshots


### Detailed test instructions:

0. Open the console and run `localStorage.setItem('debug', 'wc-admin*')` (be sure you have verbose mode activated)
1. With a completed onboarding go to Marketing -> Pinterest -> Settings Tab
2. Change some options and click save
3. You should see a debugging output showing the event `wcadmin_pfw_save_changes_button_click` with `pinterest_settings` in context prop and the options updated.
4. Change some options in Settings Tab, move to Connection Tab and save. 
5. You should see a debugging output showing the event `wcadmin_pfw_save_changes_button_click` with the properties updated. 
6. Check that reloading the page persists the saved options 
7. Check and run the unit tests with `npm run test:js` 

### Technical details

This follows the same strategy as in Analytics -> Settings -> Save. We send the context and the properties to be saved. Doesn't matter if they are actually saved in DB or not.

We only allow to track the properties registered in `TRACKABLE_SETTINGS` inside `prepare-for-tracking.js` this is for preventing `account_info` or any other current and future sensitive data to be tracked by accident. 

Regarding to context; I'm triggering `pinterest_settings` for both Settings and Connection. Since there is an issue with naming there. I'm not sure if we should add both  `pinterest_settings` and  `pinterest_connection` or only  `pinterest_settings`

### Changelog entry

> Tweak - Add tracking on Save Changes button
